### PR TITLE
Resolve hash collisions in content-addressable schedule keys

### LINKED ISSUE #1103: Compare scheduled txn bytes in case of hash collision
**Summary of the defect**
If two `ScheduleCreate` txns are created with different scheduled bytes `X` and `Y` such that `Arrays.hashCode(X) == Arrays.hashCode(Y)`, the [`HederaScheduleStore.lookup` method](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/store/schedule/HederaScheduleStore.java#L214) will incorrectly treat them as the same scheduled entity.

Suppose the txn with `X` reaches consensus first, and then `Y`.

Unless `X` and `Y` require signatures from the same keys(!), the second `ScheduleCreate` can incorrectly resolve to `NO_NEW_VALID_SIGNATURES`.

If `X` and `Y` require signatures from the same keys, and the second `ScheduleCreate` included valid signatures for one or more of these, then the first `ScheduleCreate` may incorrectly receive additional signatories.

**Suggested resolution**
Replace the `CompositeKey` with a `ContentAddressableSchedule` value object that contains the scheduled bytes as a `ByteString`.

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Related issue(s)**:
 - Closes #1103

**Summary of the change**:
- The `CompositeKey` class depended on a scheduled transaction's bytes only through the result of `Arrays.hashCode()`.
- Remove this type, and replace with a `ContentAddressableSchedule` value type that includes the transaction bytes as a `ByteString` used to meet the `Object` contract.

### CODE REVIEW

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM

**Review by qnswirlds — COMMENTED** [review]
LGTM overall. However, I am concerned by the direct access of some member variables, especially the `existingSchedules` in the `HederaScheduleStore`.

File: hedera-node/src/test/java/com/hedera/services/store/schedule/HederaScheduleStoreTest.java [hunk 1]
```diff
@@ -431,19 +429,42 @@ public void rejectsCreateProvisionallyMissingSchedulingAccount() {
...
+                extant);
+        subject.existingSchedules.put(extantKey, fromScheduleId(created));
+        // and:
+        given(schedule.transactionBody()).willReturn(extant);
+
+        // when:
+        var optionalId = subject.lookupScheduleId(candidate, payerId, adminKey, entityMemo);
+
+        // then:
+        assertTrue(optionalId.isEmpty());
```
Comment (anighanta) [inline hedera-node/src/test/java/com/hedera/services/store/schedule/HederaScheduleStoreTest.java:465]: can we also assert that `lookupScheduleId` on `extant` returns a scheduleID ?

Comment (anighanta) [reply hedera-node/src/test/java/com/hedera/services/store/schedule/HederaScheduleStoreTest.java:465]: nvmd.. covered in other tests.

**Review by anighanta — APPROVED** [review]

**Pull request merged**